src/analysis/normed_space/lp_space.lean

```diff
@@ -628,6 +628,94 @@ end
 
 end normed_space
 
+section non_unital_normed_ring
+
+variables {I : Type*} {B : I → Type*} [Π i, non_unital_normed_ring (B i)]
+
+lemma _root_.mem_ℓp.infty_mul {f g : Π i, B i} (hf : mem_ℓp f ∞) (hg : mem_ℓp g ∞) :
+  mem_ℓp (f * g) ∞ :=
+begin
+  rw mem_ℓp_infty_iff,
+  obtain ⟨⟨Cf, hCf⟩, ⟨Cg, hCg⟩⟩ := ⟨hf.bdd_above, hg.bdd_above⟩,
+  refine ⟨Cf * Cg, _⟩,
+  rintros _ ⟨i, rfl⟩,
+  calc ∥(f * g) i∥ ≤ ∥f i∥ * ∥g i∥ : norm_mul_le (f i) (g i)
+  ...             ≤ Cf * Cg       : mul_le_mul (hCf ⟨i, rfl⟩) (hCg ⟨i, rfl⟩) (norm_nonneg _)
+                                      ((norm_nonneg _).trans (hCf ⟨i, rfl⟩))
+end
+
+instance : has_mul (lp B ∞) :=
+{ mul := λ f g, ⟨(f  * g : Π i, B i) , f.property.infty_mul g.property⟩}
+
+@[simp] lemma infty_coe_fn_mul (f g : lp B ∞) : ⇑(f * g) = f * g := rfl
+
+instance : non_unital_ring (lp B ∞) :=
+function.injective.non_unital_ring lp.has_coe_to_fun.coe (subtype.coe_injective)
+  (lp.coe_fn_zero B ∞) lp.coe_fn_add infty_coe_fn_mul lp.coe_fn_neg lp.coe_fn_sub
+  (λ _ _, rfl) (λ _ _,rfl)
+
+instance : non_unital_normed_ring (lp B ∞) :=
+{ norm_mul := λ f g, lp.norm_le_of_forall_le (mul_nonneg (norm_nonneg f) (norm_nonneg g))
+    (λ i, calc ∥(f * g) i∥ ≤ ∥f i∥ * ∥g i∥ : norm_mul_le _ _
+    ...                    ≤ ∥f∥ * ∥g∥
+    : mul_le_mul (lp.norm_apply_le_norm ennreal.top_ne_zero f i)
+        (lp.norm_apply_le_norm ennreal.top_ne_zero g i) (norm_nonneg _) (norm_nonneg _)),
+  .. lp.normed_group }
+
+-- we also want a `non_unital_normed_comm_ring` instance, but this has to wait for #13719
```

Comment:
I think this is fine here, the TODO is probably more useful here so that someone looking for the instance can immediately see where it ought to go; especially since it refers to a typeclass that doesn't even exist yet!